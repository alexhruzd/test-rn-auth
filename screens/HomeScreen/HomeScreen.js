import React, { useEffect } from "react";
import { StyleSheet, Text, View } from "react-native";


const HomeScreen = ({navigation}) => {

  useEffect(() => {
    navigation.openDrawer();
  }, []);

  return (
    <View style={styles.container}>
      <Text>Hello!! This is HomePage</Text>
    </View>
  );
};

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#fff",
  },
});

export default HomeScreen;
