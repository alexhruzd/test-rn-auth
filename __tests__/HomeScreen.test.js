import React from "react";

import { cleanup, render } from "@testing-library/react-native";

import HomeScreen from "../screens/HomeScreen";

afterEach(cleanup);

describe("HomeScreen", () => {
  it("should show Hello!! This is HomePage", () => {
    const homeText = "Hello!! This is HomePage!";
    const notFoundText = "Not found text";

    const { toJSON, getByText, queryByText } = render(<HomeScreen />);

    const foundHomeText = getByText(homeText);
    const notFoundTextElement = queryByText(notFoundText);

    expect(foundHomeText.props.children).toEqual(homeText);
    expect(notFoundTextElement).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("HomeScreen", () => {
  it("should not find a dynamic component", () => {
    const {queryByTestId} = render(<HomeScreen />);

    const notFindDynamicElement = queryByTestId("dinamic component");

    expect(notFindDynamicElement).toBeNull();
  });
});
