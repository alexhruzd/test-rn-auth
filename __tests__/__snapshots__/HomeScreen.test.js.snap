// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should show Hello!! This is HomePage 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello!! This is HomePage!
  </Text>
</View>
`;
